MCJ_37

ThomassenCathleia opened the pull request:
Adjusted the ErrorController for to be more generic
It should show an alert if it cannot connect or if the credentials are
invalid

Commit:
Adjusted the ErrorController for to be more generic

It should show an alert if it cannot connect or if the credentials are
invalid

lenshuygh reviewed (changes requested):
Nice, tested scenario's:
- no props set: OK
- no DB running: OK
- typo in username: OK
- typo in pass: OK
- typo in url-ip: OK
- typo in url-db-name -> java.sql.SQLSyntaxErrorException: Unknown database 'mcjav' if DB-name wrong
=> had to disable name/telephone checking in first screen !

ThomassenCathleia committed:
Adjusted phone number so slashes and dashes are allowed

Refactored code so an error will always exit the application.

The setter should check if it's a valid JDBC URL. If it's not and it
still passes the check, the error is caught upon connecting.

ThomassenCathleia committed:
Fixed wrong regex. Cleaned the whole telephone number string instead of
just not the numbers

ThomassenCathleia committed:
Customer names can now include a space
